drivers/adc/ad7124/ad7124.c

```diff
@@ -1037,11 +1038,14 @@ int32_t ad7124_setup(struct ad7124_dev **device,
 	ret = ad7124_read_register(dev, &dev->regs[AD7124_ID_REG]);
 	if (ret)
 		goto error_spi;
+
+	id_val = dev->regs[AD7124_ID_REG].value;
 	if (dev->active_device == ID_AD7124_4) {
-		if (dev->regs[AD7124_ID_REG].value != AD7124_4_ID)
+		if (id_val != AD7124_4_ID && id_val != AD7124_4_NEW_ID)
```

Comment:
a switch case might fit better here, especially if there will be other 'new ids' in the future or other supported devices by the same driver